VTUL/dlp-access
# LIBTD-2321: Create SiteAdmin page to update site configurations with authentication

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2321) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)
This ticket creates initial authentication page for https://webapps.es.vt.edu/jira/browse/LIBTD-2314

# What does this Pull Request do? (:star:)
This PR creates an authentication page for an user to sign up/in. If the user belongs to the "SiteAdmin" group, he/she is able to access this page; otherwise, access is denied. 

# What's the changes? (:star:)

* Changed the add_to_group function so that a signed-in user will not be automatically added to the "Admin" group
* Added route for SiteAdmin page
* Added SiteAdmin component with authentication capability and grant access by checking if the user is in the "SiteAdmin" group

# How should this be tested?

* "amplify push" to disable the automatic add-to-admin-group function
* Go to /siteAdmin page, sign up/in, if you are not in "SiteAdmin" group, no access
* Add the user to "SiteAdmin" group in AWS console, sign out and sign in again, check if access is granted

# Interested parties
Tag (@yinlinchen ) interested parties

(:star:) Required fields
